core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1723,36 +1722,6 @@ class KafkaZkClient private[zk] (
     }
   }
 
-  def getOrCreateMigrationState(initialState: ZkMigrationLeadershipState): ZkMigrationLeadershipState = {
```

Comment:
It looks like `tryRegisterKRaftControllerAsActiveController()` is no longer called, so it can be deleted too.